Tests/SourceKitLSPTests/ExecuteCommandTests.swift

```diff
@@ -136,6 +137,123 @@ final class ExecuteCommandTests: XCTestCase {
+            in context: some MacroExpansionContext
+          ) -> ExprSyntax {
+            guard let argument = node.argumentList.first?.expression else {
+              fatalError("compiler bug: the macro does not have any arguments")
+            }
+
+            return "(\(argument), \(literal: argument.description))"
+          }
+        }
+
+        @main
+        struct MyMacroPlugin: CompilerPlugin {
+          let providingMacros: [Macro.Type] = [
+            StringifyMacro.self,
+          ]
+        }
+        """#,
+        "MyMacroClient/MyMacroClient.swift": """
+        @freestanding(expression)
+        public macro stringify<T>(_ value: T) -> (T, String) = #externalMacro(module: "MyMacros", type: "StringifyMacro")
+
+        func test() {
+          1️⃣#2️⃣stringify3️⃣(1 + 2)
+        }
+        """,
+      ],
+      manifest: SwiftPMTestProject.macroPackageManifest
+    )
+    try await SwiftPMTestProject.build(at: project.scratchDirectory)
+
+    let (uri, positions) = try project.openDocument("MyMacroClient.swift")
+
+    let positionMarkersToBeTested = [
+      (start: "1️⃣", end: "1️⃣"),
+      (start: "2️⃣", end: "2️⃣"),
+      (start: "1️⃣", end: "3️⃣"),
+      (start: "2️⃣", end: "3️⃣"),
+    ]
+
+    for positionMarker in positionMarkersToBeTested {
+      let args = ExpandMacroCommand(
+        positionRange: positions[positionMarker.start]..<positions[positionMarker.end],
+        textDocument: TextDocumentIdentifier(uri)
+      )
+
+      let metadata = SourceKitLSPCommandMetadata(textDocument: TextDocumentIdentifier(uri))
+
+      var command = args.asCommand()
+      command.arguments?.append(metadata.encodeToLSPAny())
+
+      let request = ExecuteCommandRequest(command: command.command, arguments: command.arguments)
+
+      let expectation = XCTestExpectation(description: "Handle Show Document Request")
+      let showDocumentRequestURI = ThreadSafeBox<DocumentURI?>(initialValue: nil)
+
+      project.testClient.handleSingleRequest { (req: ShowDocumentRequest) in
+        showDocumentRequestURI.value = req.uri
+        expectation.fulfill()
+        return ShowDocumentResponse(success: true)
+      }
+
+      let result = try await project.testClient.send(request)
+
+      guard let resultArray: [MacroExpansionEdit] = Array(fromLSPArray: result ?? .null) else {
+        XCTFail(
+          "Result is not an array. Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
+        )
+        return
+      }
+
+      XCTAssertEqual(
+        resultArray.count,
+        1,
+        "resultArray is empty. Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
+      )
+      XCTAssertEqual(
+        resultArray.only?.newText,
+        "(1 + 2, \"1 + 2\")",
+        "Wrong macro expansion. Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
+      )
+
+      await fulfillment(of: [expectation], timeout: 10.0)
```

Comment:
`fulfillmentOfOrThrow` is a useful utility that uses `defaultTimeout` and throws an error if it times out, which is useful because it’s very likely that all the following assertions will fail if the expectation times out.

```suggestion
      try await fulfillmentOfOrThrow([expectation])
```